charts/argo-cd/templates/argocd-configs/argocd-secret.yaml

```diff
@@ -0,0 +1,28 @@
+{{- if .Values.configs.secret.createSecret }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: argocd-secret
+  labels:
+    release: {{ .Release.Name }}
+    app.kubernetes.io/name: {{ include "argo-cd.name" . }}-secret
+    helm.sh/chart: {{ include "argo-cd.chart" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/part-of: {{ include "argo-cd.name" . }}
+    app.kubernetes.io/component: {{ .Values.server.name }}
+type: Opaque
+{{- if or .Values.configs.secret.githubSecret (or .Values.configs.secret.gitlabSecret .Values.configs.secret.bitbucketSecret) }}
+# Setting a blank data again will wipe admin password/key/cert
```

Comment:
Should we support passing in the admin password here?

Reply:
I'd say yes, it's a common enough pattern that folks want to set it and the entire secret can be done out of band if more security is needed.